src/modules/acbs/dto/acbs-party-external-ratings-response.dto.ts

```diff
@@ -0,0 +1,23 @@
+export type AcbsPartyExternalRatingsResponseDto = AcbsPartyExternalRatingDto[];
+
+interface AcbsPartyExternalRatingDto {
+  PartyIdentifier: string;
+  RatingEntity: {
+    RatingEntityCode: string;
+  };
+  AssignedRating: {
+    AssignedRatingCode: string;
+  };
+  RatedDate: Date; // TODO APIM-76: axios will actually return a string here, should we transform this with an axios interceptor?
```

Comment:
Everything _works_ at the moment because we don't interact with the rated date at all, but if we ever wanted to interact with it then we might run into issues because:
- we're expecting it to be a date,
- but axios / the http service will leave it as a string (that just happens to be a date time string)

I think we should add an axios interceptor to transform date time strings into dates

Reply:
1. Do we expect to interact with it as part of the POST endpoint?
2. can you give me an example as to what it's returned now and how it should be with interceptors?

Reply:
1. It doesn't look like we interact with it as part of the POST endpoint. There are some endpoints that do transform datetime string response fields though, e.g. the `GET /party?searchText={...}` endpoint converts the datetime string from ACBS into a date only string for our response

2. Currently the `HttpService` from `nestjs/axios` will return a response object with a field called `RatedDate` that is a `string` (e.g. `"2023-02-01T00:00:00Z"`). If we added an axios interceptor, we could have it regex match any string fields that are in datetime format in HTTP responses and automatically parse them to a JS `Date` object. Then the `HttpService` would instead return a `RatedDate` that is a `Date` (e.g. `new Date("2023-02-01T00:00:00Z")`).

3. Alternatively we could try to fix the problem without parsing the datetime string, e.g. we could introduce an alias type, `type DateString = string;`, and use this whenever there's a date string response field that we're not parsing, and tell Swagger that it should document any field we return with `DateString` type as actually being of `Date` type

I think the "parsing with an interceptor" approach (2) makes our life easier for later and will be more work if we have to implement it late in the project rather than early. Alternatively, if we want to go with the least work option for now, then we could use an approach like (3) until we find out if we really need (2)

Reply:
I'm happy to go for the solution that replicates the existing functionality

Reply:
I've gone for option 3 for now, we can decide if we want to parse dates automatically later